app/services/user_signup.rb

```diff
@@ -0,0 +1,42 @@
+class UserSignup 
+  attr_accessor :user, :status, :error_message
+
+  def initialize(user)
+    @user = user
+  end 
+
+  def sign_up(stripe_token, invitation_token)
+    if user.valid?
+      handle_invitation(invitation_token)
```

Comment:
In real life you might want to handle the invitation after the charge was successful.